Library/ViewModels/DiscoveryViewModel.swift

```diff
@@ -155,17 +162,39 @@ public final class DiscoveryViewModel: DiscoveryViewModelType, DiscoveryViewMode
 
     self.sortsAreEnabled = self.setSortsEnabledProperty.signal.skipNil()
 
-    currentParams
-      .takePairWhen(self.sortPagerSelectedSortProperty.signal.skipNil().skipRepeats(==))
-      .observeValues {
-        AppEnvironment.current.ksrAnalytics.trackDiscoverySelectedSort(nextSort: $1, params: $0)
-      }
+    Signal.combineLatest(
+      currentParams,
+      self.sortPagerSelectedSortProperty.signal.skipNil().skipRepeats(==),
+      prevSortedPage
+    )
+    .filter { $0.1.rawValue != $0.2.rawValue }
+    .observeValues { currentParams, currentSortedPage, prevSortedPage in
+      AppEnvironment
+        .current
+        .ksrAnalytics
+        .trackDiscoverySelectedSort(
```

Comment:
Could you explain why we are calling the event in two separate blocks with slightly different input signals?

Reply:
I can move them in one block using `merge`, but I kind of have a fear totally changing what currently exist. I'll do that now, and you can review.